### ISSUE TITLE
Switching between build and build -d takes lots of time

### ISSUE THREAD

**Issue by arekzaluski** [ARMmbed/yotta#651]
Switching between build for deployment and debugging takes a long time.
It is a relatively common scenario to change between deploy and debug.

**Comment by thegecko** [comment]
Is this because the build only caches the last build configuration which subsequently changes with additional switches?

**Comment by autopulated** [comment]
Yeah: one way to improve this would be to have separate build directories for optimised / non-optimised builds, like we do for different targets.
